### ISSUE TITLE
[stt] Better default value for maxSilenceSeconds / inactivityTimeout setting

### ISSUE THREAD

**Issue by lolodomo** [openhab/openhab-addons#12451]
The default value for maxSilenceSeconds setting is 5 seconds in the Google STT and Vosk STT.
For the Watson STT, the similar setting is named inactivityTimeout and its default value is 3 seconds.

5 seconds is largely too much in practice.
I would suggest to switch to 2 seconds, max 3 seconds.

Additionaly, it could be good to rename the setting in the Watson STT service to have the same setting name in all STT services.

@GiviMAD : WDYT ?

**Comment by GiviMAD** [comment]
The reason behind this is that those services (Google STT and Vosk STT) support the single utterance mode which made that you don't need to wait the maxSilenceSeconds for the transcription to stop so having a large value there just allows you to wait in silence before start talking.

But yes, I think that unify the option names and default values for does services makes sense and also I will try to add support for the single utterance mode in Watson STT.

I will assign the issue to myself and check to check it in the afternoon.

**Comment by lolodomo** [comment]
> The reason behind this is that those services (Google STT and Vosk STT) support the single utterance mode which made that you don't need to wait the maxSilenceSeconds for the transcription to stop so having a larger value there just allows you to wait in silence before start talking.

Ok. I am using the default settings so singleUtteranceMode is true and I noticed a very long delay before SST ended. I thought it was due to maxSilenceSeconds but I understand now that it is not. 
Do you notice this big delay too? 
Does it mean singleUtteranceMode is not a good mode?

**Comment by lolodomo** [comment]
> And no I didn't face a long delay, how long are we talking?

I can measure it more precisely this evening but I would say around 5 seconds

**Comment by GiviMAD** [comment]
> I can measure it more precisely this evening but I would say around 5 seconds

No, I'm not facing this problem tested with the latest openHAB snapshot and vosk. With both the javasound source and the pulse audio source (last version from the active pr).
I think is just the time that vosk takes to transcript the audio (which can be different based on the model and the computer), you should be able to check if this is the case by enabling the debug logs on the vosk add-on.

**Comment by lolodomo** [comment]
Here are logs with timings:
```
22:45:19.567 [DEBUG] [.core.voice.internal.VoiceManagerImpl] - Executing a simple dialog for source Entrée audio Sony Eye (pulseaudio:source:rpi2:entree)
22:45:19.571 [DEBUG] [ternal.PulseaudioSimpleProtocolStream] - Simple TCP Stream connecting
22:45:19.601 [DEBUG] [ternal.PulseaudioSimpleProtocolStream] - Scheduling disconnect
22:45:20.372 [DEBUG] [b.core.voice.internal.DialogProcessor] - RecognitionStartEvent event received
22:45:20.377 [DEBUG] [voice.voskstt.internal.VoskSTTService] - Partial: {
  "partial" : ""
}
22:45:20.382 [DEBUG] [voice.voskstt.internal.VoskSTTService] - Partial: {
  "partial" : ""
}
22:45:20.386 [DEBUG] [voice.voskstt.internal.VoskSTTService] - Partial: {
  "partial" : ""
}
22:45:22.021 [DEBUG] [voice.voskstt.internal.VoskSTTService] - Partial: {
  "partial" : ""
}
22:45:22.027 [DEBUG] [voice.voskstt.internal.VoskSTTService] - Partial: {
  "partial" : ""
}
22:45:22.102 [DEBUG] [voice.voskstt.internal.VoskSTTService] - Partial: {
  "partial" : ""
}
22:45:22.106 [DEBUG] [voice.voskstt.internal.VoskSTTService] - Partial: {
  "partial" : ""
}
22:45:22.179 [DEBUG] [voice.voskstt.internal.VoskSTTService] - Partial: {
  "partial" : ""
}
22:45:22.185 [DEBUG] [voice.voskstt.internal.VoskSTTService] - Partial: {
  "partial" : ""
}
22:45:22.189 [DEBUG] [voice.voskstt.internal.VoskSTTService] - Partial: {
  "partial" : ""
}
22:45:24.763 [DEBUG] [voice.voskstt.internal.VoskSTTService] - Partial: {
  "partial" : ""
}
22:45:25.452 [DEBUG] [voice.voskstt.internal.VoskSTTService] - Partial: {
  "partial" : ""
}
22:45:25.460 [DEBUG] [voice.voskstt.internal.VoskSTTService] - Partial: {
  "partial" : ""
}
22:45:25.469 [DEBUG] [voice.voskstt.internal.VoskSTTService] - Partial: {
  "partial" : ""
}
22:45:26.715 [DEBUG] [voice.voskstt.internal.VoskSTTService] - Partial: {
  "partial" : "allume"
}
22:45:26.984 [DEBUG] [voice.voskstt.internal.VoskSTTService] - Partial: {
  "partial" : "allume la"
}
22:45:26.996 [DEBUG] [voice.voskstt.internal.VoskSTTService] - Partial: {
  "partial" : "allume la"
}
22:45:27.002 [DEBUG] [voice.voskstt.internal.VoskSTTService] - Partial: {
  "partial" : "allume la"
}
22:45:27.419 [DEBUG] [voice.voskstt.internal.VoskSTTService] - Partial: {
  "partial" : "allume la lumière"
}
22:45:27.426 [DEBUG] [voice.voskstt.internal.VoskSTTService] - Partial: {
  "partial" : "allume la lumière"
}
22:45:27.717 [DEBUG] [voice.voskstt.internal.VoskSTTService] - Partial: {
  "partial" : "allume la lumière"
}
22:45:27.726 [DEBUG] [voice.voskstt.internal.VoskSTTService] - Partial: {
  "partial" : "allume la lumière"
}
22:45:28.111 [DEBUG] [voice.voskstt.internal.VoskSTTService] - Partial: {
  "partial" : "allume la lumière"
}
22:45:28.117 [DEBUG] [voice.voskstt.internal.VoskSTTService] - Partial: {
  "partial" : "allume la lumière"
}
22:45:28.423 [DEBUG] [voice.voskstt.internal.VoskSTTService] - Partial: {
  "partial" : "allume la lumière"
}
22:45:28.430 [DEBUG] [voice.voskstt.internal.VoskSTTService] - Partial: {
  "partial" : "allume la lumière"
}
22:45:28.437 [DEBUG] [voice.voskstt.internal.VoskSTTService] - Partial: {
  "partial" : "allume la lumière"
}
22:45:28.547 [DEBUG] [voice.voskstt.internal.VoskSTTService] - Partial: {
  "partial" : "allume la lumière chambre"
}
22:45:28.555 [DEBUG] [voice.voskstt.internal.VoskSTTService] - Partial: {
  "partial" : "allume la lumière chambre"
}
22:45:28.661 [DEBUG] [voice.voskstt.internal.VoskSTTService] - Partial: {
  "partial" : "allume la lumière chambre"
}
22:45:28.670 [DEBUG] [voice.voskstt.internal.VoskSTTService] - Partial: {
  "partial" : "allume la lumière chambre"
}
22:45:28.745 [DEBUG] [voice.voskstt.internal.VoskSTTService] - Partial: {
  "partial" : "allume la lumière chambre"
}
22:45:28.751 [DEBUG] [voice.voskstt.internal.VoskSTTService] - Partial: {
  "partial" : "allume la lumière chambre"
}
22:45:28.826 [DEBUG] [voice.voskstt.internal.VoskSTTService] - Partial: {
  "partial" : "allume la lumière chambre"
}
22:45:28.834 [DEBUG] [voice.voskstt.internal.VoskSTTService] - Partial: {
  "partial" : "allume la lumière chambre"
}
22:45:28.840 [DEBUG] [voice.voskstt.internal.VoskSTTService] - Partial: {
  "partial" : "allume la lumière chambre"
}
22:45:28.942 [DEBUG] [voice.voskstt.internal.VoskSTTService] - Result: {
  "text" : "allume la lumière chambre"
}
22:45:28.946 [DEBUG] [b.core.voice.internal.DialogProcessor] - RecognitionStopEvent event received
22:45:28.950 [DEBUG] [voice.voskstt.internal.VoskSTTService] - Final: allume la lumière chambre
22:45:28.954 [DEBUG] [b.core.voice.internal.DialogProcessor] - SpeechRecognitionEvent event received
22:45:28.958 [DEBUG] [b.core.voice.internal.DialogProcessor] - Text recognized: allume la lumière chambre
22:45:28.965 [DEBUG] [b.core.voice.internal.DialogProcessor] - Interpretation result: Ok.
22:45:28.981 [INFO ] [openhab.event.ItemCommandEvent       ] - Item 'LampesHueChambre' received command ON
22:45:29.008 [DEBUG] [seaudio.internal.ConvertedInputStream] - Duration of input stream : 1535
22:45:29.013 [DEBUG] [seaudio.internal.ConvertedInputStream] - Sound is a MP3. Trying to reencode it
22:45:29.034 [DEBUG] [seaudio.internal.ConvertedInputStream] - Sound is not in the target format. Trying to reencode it
22:45:29.039 [DEBUG] [ternal.PulseaudioSimpleProtocolStream] - Simple TCP Stream connecting
22:45:29.135 [DEBUG] [lseaudio.internal.PulseAudioAudioSink] - Sleep time to let the system play sound : 1449
22:45:30.589 [DEBUG] [ternal.PulseaudioSimpleProtocolStream] - Scheduling disconnect
22:45:30.637 [DEBUG] [ternal.PulseaudioSimpleProtocolStream] - Scheduling disconnect
22:45:30.643 [WARN ] [eaudio.internal.PulseAudioAudioSource] - IOException while writing to from pulse source pipe: null
```

As you can see, around 8 seconds between RecognitionStartEvent  and RecognitionStopEvent. An initial second would be expected as I am talking not immediately after launching the command.
It looks like that Vosk is detecting with a delay.

Interesting is also the last lines showing a reencoding of the audio. Do you think it could be the cause of this delay ? By the way, I should try to update the default framerate in the remote pulseaudio server.

And finally, you can se an IOException  with null source pipe.

You can also see a success of my vocal command ;)

**Comment by GiviMAD** [comment]
```
22:45:30.643 [WARN ] [eaudio.internal.PulseAudioAudioSource] - IOException while writing to from pulse source pipe: null
```
Hope that with the logs output changes in the pulseaudio source those errors will give more info.

Yes, seems like some kind of delay inside Vosk, I don't think is related to the reencode of the audio.

```
And finally, you can se an IOException with null source pipe.
```
But the "Pipe not open" error seems to be solved with the latests changes, or do you still see any of those?
Maybe the underling IOException is just an interruption exception that can be ignored.

**Comment by lolodomo** [comment]
I was running your very last version of Pulseaudio (to be exact, a version with my own changes to make it compiling).
So there is unfortunately no additional info.

Yes, I don't encounter anymore the  "Pipe not open" error.

I would say that now the only concrete problem is that sometimes Vosk is detecting nothing, as if it was not really listening the audio source. This looks like something new I did not encounter before. It looks like it replaces the old problem with exception. It happens frequently.

**Comment by lolodomo** [comment]
Audio reencoding concerns TTS and not STT, correct?

**Comment by GiviMAD** [comment]
> Audio reencoding concerns TTS and not STT, correct?

Yes.

> I would say that now the only concrete problem is that sometimes Vosk is detecting nothing.

Are you sure you are not talking to soon?
Sounds stupid but it happens to me while testing. As the STT takes some time to be ready, you can not talk just after the spot happens, you need to wait until the recognition start event is fired.

**Comment by lolodomo** [comment]
> Are you sure you are not talking to soon?
> Sounds stupid but it happens to me while testing. As the STT takes some time to be ready, you can not talk just after the spot happens, you need to wait until the recognition start event is fired.

Looking at the logs I published, it takes 800 ms to get the start event. 
I believe I waited at least this delay before speaking but I will try agsin.

**Comment by lolodomo** [comment]
Closed by #12453
